src/common/iso_8601.cc

```diff
@@ -41,7 +41,8 @@ optional<real_time> calculate(const tm& t, uint32_t n = 0) {
     return none;
   }
 
-  return real_clock::from_time_t(tt) + nanoseconds(n);
+  return boost::make_optional<real_time>(real_clock::from_time_t(tt)
+                                         + nanoseconds(n));
 }
```

Comment:
@tchaikov 
That could very well be the case. My system Boost is at version 1.64.